### PR TITLE
Argo CD Contribution

### PR DESCRIPTION
**Pull request by huytn1219**
#### What this PR does / why we need it:
Add ArgoCD lab under Automation section

#### Which issue(s) this PR fixes:
https://github.com/aws-samples/eks-workshop-v2/issues/319

Fixes #

#### Quality checks

- [+] My content adheres to the style guidelines
- [+] I ran `make test` or `make e2e-test` and it was successful

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: website/docs/automation/gitops/argocd/update.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+In this lab excerise, we'll update the `catalog` application to increase replicas to handle higher traffic with Argo CD. 
+
+Update replicas of the `catalog` application in the `kuztomization` file to 3:
+
+```file
+automation/gitops/argocd/update-app/kustomization.yaml
+```
+
+:::tip In your environment
+You should update the same `kustomization.yaml` file and commit to the Git repository.
```
Comment (allamand) [inline website/docs/automation/gitops/argocd/update.md:17]: I would rephrase with something like:

Normally, you should update your files and commit in the Git repository. In this lab exercice, we provide you with a new version of the file to simulate that we have made a change, and we will instead update the path for our argocd-demo application with the new file.

**Review by allamand — CHANGES_REQUESTED** [review]
Hi @huytn1219, Thanks you so much for your contribution.

Can you take a look on some of the comments I put in the PR.

Sébastien

File: terraform/modules/cluster/addons.tf [hunk 2]
```diff
@@ -236,6 +237,22 @@ module "eks_blueprints_kubernetes_addons" {
     additional_irsa_policies = ["arn:aws:iam::aws:policy/AdministratorAccess"]
   }
 
+  argocd_helm_config = {
```
[the remark this answers is not in the document]
Comment (huytn1219) [reply terraform/modules/cluster/addons.tf:240]: @allamand Thank you for your feedbacks. I resolved most of the recommendation. Please have a look when you can. The only thing I haven't done is this one. I'll wait for @niallthomson or @csantanapr for the suggestions.

File: website/docs/automation/gitops/argocd/ui.md [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 sidebar_position: 10
 weight: 10
 ---
- 
-For the purpose of this lab, Argo CD server has already been exposed outside of the cluster using Load Balancer type.
 
-Get the URL from Argo CD service resource:
+:::info
+For the purpose of this lab, Argo CD server has already been exposed outside of the cluster using `Load Balancer` type. To see how we set up Argo CD in the cluster with EKS Terraform Blueprint, please refer to this [section](https://github.com/aws-samples/eks-workshop-v2/blob/d12e0008d688460f2d0a93c83e1ff3c5b03f72e6/terraform/modules/cluster/addons.tf#L240-L254).
+:::
```
Comment (huytn1219) [inline website/docs/automation/gitops/argocd/ui.md:9]: The URL needs to be updated later to reflect the correct files and lines number.

File: website/docs/automation/gitops/argocd/deploy.md [hunk 4]
```diff
@@ -24,16 +24,24 @@ $ kubectl create ns argocd-demo
 Create an Argo CD application:
 
 ```bash
-$ argocd app create argocd-demo --repo https://github.com/aws-samples/eks-workshop-v2.git --path environment/workspace/manifests/catalog --dest-server https://kubernetes.default.svc --dest-namespace argocd-demo
+$ argocd app create argocd-demo --repo https://github.com/aws-samples/eks-workshop-v2.git --path environment/workspace/modules/automation/gitops/argocd --dest-server https://kubernetes.default.svc --dest-namespace argocd-demo
 $ application 'argocd-demo' created
```
Comment (allamand) [inline website/docs/automation/gitops/argocd/deploy.md:28]: You should remove the $ on the second line so that it is not copy

**Review by allamand — CHANGES_REQUESTED** [review]
Thanks so much, i've just ask a minor change on a command copy i forgott in the previous review.

File: website/docs/automation/gitops/argocd/deploy.md (hunk 4 above)
Comment (huytn1219) [reply website/docs/automation/gitops/argocd/deploy.md]: good catch! I removed the $ sign

**Review by allamand — DISMISSED** [review]
Thanks for this module.

I tested it and all went good. Thanks for your contribution

**Comment by allamand** [comment]
Tests for this module are missing, will be done in https://github.com/aws-samples/eks-workshop-v2/issues/367

**Comment by huytn1219** [comment]
> 

Thanks @allamand

File: terraform/modules/cluster/addons.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (niallthomson) [reply terraform/modules/cluster/addons.tf:240]: For pretty much all the addons we try to scale down the number of replicas to 1.

**Comment by allamand** [comment]
@huytn1219 can you close and reopen the PR ? We have changed some part of the automation that should re-trigger the tests. Thanks

**Comment by huytn1219** [comment]
@allamand looks like the test is still failing

**Comment by allamand** [comment]
Thanks @niallthomson can you look at this ?

**Comment by allamand** [comment]
@huytn1219 can you rebase ?

**Comment by niallthomson** [comment]
Since we have an urgent need for this module I'm going to merge it then make some follow up fixes. Thank you for the work on this content!
